### PR TITLE
Endpoint to retrieve the last uploaded CVs #95

### PR DESCRIPTION
**Pull request by CharlesKimMaina**
# Title :  Endpoint to retrieve the last uploaded CVs #95

One can  retrieve from the DB a list the 3 last upload CVs or All the whole list (limit to 20), so that you can view them on Frontend. 

Fixes # https://github.com/HackYourFuture-CPH/rate-my-cv/issues/95

# How to test?
On frontend, you can see the 3 last uploaded CVs as below; 
last3uploadedCvs

On backend Testing,., I joined users and cv tables, I created new usercv router and usercv controller. I also limit the cvs numbers to be fetched from backend to max 20 cvs. see joined table below; 
users+cvs_JOINED_tables

# Checklist

- [*] I have performed a self-review of my own code
- [*] I have commented my code, particularly in hard-to-understand areas
- [*] I have made corresponding changes to the documentation
- [*] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/components/RetrivingCvs/RetrivingCvs.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from 'react';
+import { CardCvViewSearchComponent } from '../CardCvViewSearchComponent/CardCvViewSearchComponent';
+
+export default function RetriveCvs() {
+  const [cvs, setCvs] = useState([]);
+  useEffect(() => {
+    (async () => {
+      const result = await fetch('/api/usercv').then((res) => res.json());
+      console.log(result);
```
Comment (bytewiz) [inline src/client/components/RetrivingCvs/RetrivingCvs.js:9]: lets remove console.logs

File: src/client/components/RetrivingCvs/RetrivingCvs.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      setCvs(result);
+    })();
+  }, []);
+  console.log(cvs);
```
Comment (bytewiz) [inline src/client/components/RetrivingCvs/RetrivingCvs.js:13]: lets remove console.logs

File: src/client/components/RetrivingCvs/RetrivingCvs.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from 'react';
+import { CardCvViewSearchComponent } from '../CardCvViewSearchComponent/CardCvViewSearchComponent';
+
+export default function RetriveCvs() {
```
Comment (bytewiz) [inline src/client/components/RetrivingCvs/RetrivingCvs.js:4]:
```suggestion
export default function RetrieveCvs() {
```

File: src/client/components/RetrivingCvs/RetrivingCvs.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    (async () => {
+      const result = await fetch('/api/usercv').then((res) => res.json());
+      console.log(result);
+      setCvs(result);
+    })();
+  }, []);
+  console.log(cvs);
+  return (
+    <div>
+      {cvs.slice(-3).map((item) => (
```
Comment (bytewiz) [inline src/client/components/RetrivingCvs/RetrivingCvs.js:16]: Why is this sliced?

**Review by bytewiz — CHANGES_REQUESTED** [review]

File: src/client/components/RetrivingCvs/RetrivingCvs.js (hunk 2 above)
Comment (CharlesKimMaina) [reply src/client/components/RetrivingCvs/RetrivingCvs.js:13]: okey, thanks ,.,i will remove all console.logs

File: src/client/components/RetrivingCvs/RetrivingCvs.js (hunk 4 above)
Comment (CharlesKimMaina) [reply src/client/components/RetrivingCvs/RetrivingCvs.js:16]: to upload only 3 last uploaded cvs,. The original array will not be modified, but the task requires that we show only 3 latest cvs,.,Thats why i use slice like in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice ,.,.,under parameters

Comment (KavyaReddy732) [reply src/client/components/RetrivingCvs/RetrivingCvs.js:14]: you can use conditions instead by using array length and boolean value

**Review by kamalrsa — COMMENTED** [review]
I can see that .slice (-3) working fine to isplay last 3 cvs

**Comment by CharlesKimMaina** [comment]
> I can see that .slice (-3) working fine to isplay last 3 cvs

**Review by dpfernandes — CHANGES_REQUESTED** [review]
Please remove the Frontend changes on this PR. 

This PR should be only to add the endpoint code

File: src/server/api/routes/usercv.router.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+const router = express.Router({ mergeParams: true });
+
+// controllers
+const usercvController = require('../controllers/usercv.controller');
+
+/**
+ * @swagger
+ * /modules:
+ *  get:
+ *    summary: Get all cvs and users
```
Comment (pdsorensen) [inline src/server/api/routes/usercv.router.js:14]: Maybe it should say that there is a limit on 20?

File: src/server/api/controllers/usercv.controller.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+/*  This is the controller for cvs */
+
+const knex = require('../../config/db');
+
+const getCvsnUsers = async () => {
```
Comment (pdsorensen) [inline src/server/api/controllers/usercv.controller.js:5]: i think `getUserCvs`  makes more sense.

**Review by pdsorensen — CHANGES_REQUESTED** [review]
Nice one, only have some minor suggestions.

File: src/server/api/routes/usercv.router.js (hunk 5 above)
Comment (CharlesKimMaina) [reply src/server/api/routes/usercv.router.js:14]: Sure, Patrick,.let me change on swagger documentation, Thanks

File: src/server/api/controllers/usercv.controller.js (hunk 6 above)
Comment (CharlesKimMaina) [reply src/server/api/controllers/usercv.controller.js:5]: true, let me change it a push aNew,.,

**Review by pdsorensen — APPROVED** [review]

**Pull request merged**